spec/models/procedure_spec.rb

```diff
@@ -536,38 +536,77 @@
 
     after { Timecop.return }
 
-    context "without parent procedure" do
+    context "without canonical procedure" do
       before do
         Timecop.freeze(now)
         procedure.publish!
       end
 
       it do
+        expect(procedure.canonical_procedure).to be_nil
         expect(procedure.closed_at).to be_nil
         expect(procedure.published_at).to eq(now)
         expect(Procedure.find_by(path: "example-path")).to eq(procedure)
         expect(Procedure.find_by(path: "example-path").administrateurs).to eq(procedure.administrateurs)
       end
     end
+
+    context "with canonical procedure" do
+      let(:canonical_procedure) { create(:procedure, :published) }
+
+      before do
+        Timecop.freeze(now)
+        procedure.publish!(canonical_procedure)
+      end
+
+      it do
+        expect(procedure.canonical_procedure).to eq(canonical_procedure)
+        expect(procedure.closed_at).to be_nil
+        expect(procedure.published_at).to eq(now)
+      end
+    end
   end
 
   describe "#publish_or_reopen!" do
-    let(:published_procedure) { create(:procedure, :published) }
-    let(:administrateur) { published_procedure.administrateurs.first }
+    let(:canonical_procedure) { create(:procedure, :published) }
+    let(:administrateur) { canonical_procedure.administrateurs.first }
 
     let(:procedure) { create(:procedure, administrateurs: [administrateur]) }
     let(:now) { Time.zone.now.beginning_of_minute }
 
-    context "without parent procedure" do
+    context "without canonical procedure" do
+      before do
+        Timecop.freeze(now)
```

Comment:
J'ai l'impression que ton test n'appelle pas `Timecop.return`.

Cela dit on pourrait juste faire ça, non ?

```suggestion
        procedure.path = canonical_procedure.path
        Timecop.freeze(now) do
          procedure.publish_or_reopen!(administrateur)
        end
        canonical_procedure.reload
```